download_plugins.py

```diff
@@ -92,72 +96,83 @@ def download_plugin(plugin, version=None):
 # Install each plugin in the supplied json file along with dependencies.
 def install_plugins():
     for plugin, version in plugins.items():
-        print("\n**** Install Plugin: {} *****".format(plugin))
-        download_plugin(plugin, version)
+        debug("\n**** Add Plugin: {} *****".format(plugin))
+        if version is None:
+            version = plugins_list["plugins"][plugin]["version"]
+        add_dep(plugin, version, plugin)
         get_dependencies(plugin)
 
-    print("\nDownloading all dependencies")
-    print("*****************************************")
+    debug("\nDownloading all dependencies")
+    debug("*****************************************")
     for plugin, version in stored_plugins.items():
         download_plugin(plugin, version)
 
-    for dependency, info in dependencies_info.items():
+    for dependency, info in dep_info.items():
         if not (info["duplicate"]):
             continue
         print(dedent("""\
         Warning: Multiple versions of {} found in dependencies.\
         """.format(dependency)))
         for parent in info["parents"]:
             for version, plugin in parent.items():
-                print("    Version {version} required by {plugin}".format(
-                    version=version, plugin=plugin
-                ))
+                if dependency == plugin:
+                    print("    !! TOP-LEVEL version: {} !!".format(version))
+                else:
+                    print("    Version {version} required by {plugin}".format(
+                        version=version, plugin=plugin
+                    ))
```

Comment:
As we discussed in person, could you put each warning message under the downloaded information? For example, this is how the output looks like right now:

```
...
Downloaded workflow-support: 2.17
Downloaded workflow-job: 2.21 [TOP-LEVEL]
Downloaded workflow-basic-steps: 2.6
...
Warning: Multiple versions of blah found in dependencies.
...
Warning: Multiple versions of workflow-job found in dependencies.
    Version 2.0 required by pipeline-stage-view
    Version 2.0 required by pipeline-rest-api
    Version 2.9 required by workflow-aggregator
    Version 2.11 required by pipeline-graph-analysis
    Version 2.15 required by pipeline-model-extensions
    !! TOP-LEVEL version: 2.17 !!
    Version 2.21 required by workflow-multibranch
    Downloaded version 2.21
...
```

Could we change it to be like:

```
...
Downloaded workflow-support: 2.17
Downloaded workflow-job: 2.21 [TOP-LEVEL]
    Warning: Multiple versions of workflow-job found in dependencies.
        Version 2.0 required by pipeline-stage-view
        Version 2.0 required by pipeline-rest-api
        Version 2.9 required by workflow-aggregator
        Version 2.11 required by pipeline-graph-analysis
        Version 2.15 required by pipeline-model-extensions
        !! TOP-LEVEL version: 2.17 !!
        Version 2.21 required by workflow-multibranch
        Downloaded version 2.21
Downloaded workflow-basic-steps: 2.6
    Warning: Multiple versions of blah found in dependencies.
...
```

Reply:
Ok, the output should now look like this without the `Downloaded version <version>` line since this information is already given in the first `Downloaded <plugin>: <version>` line.